### ISSUE TITLE
[Bug] not really, CVE-2019-8341 advice please

### ISSUE THREAD

**Issue by brabster** [dbt-labs/dbt-core#10250]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

safety reporting a vulnerability in Jinja2 transitive for dbt-core et al.

-> Vulnerability found in jinja2 version 3.1.4
   Vulnerability ID: 70612
   Affected spec: >=0
   ADVISORY: In Jinja2, the from_string function is prone to Server Side Template Injection (SSTI) where it takes the "source" parameter as
   a template object, renders it, and then returns it. The attacker can exploit it with {{INJECTION COMMANDS}} in a URI. NOTE: The maintainer...
   CVE-2019-8341
   For more information about this vulnerability, visit https://data.safetycli.com/v/70612/97c
   To ignore this vulnerability, use PyUp vulnerability id 70612 in safety’s ignore command-line argument or add the ignore to your safety policy
   file.

### Expected Behavior

No vulnerabilities reported, ideally. There's no fix version available (and might never be) so this is more to have you confirm that dbt-core is not affected by this vuln. I don't think it is as I don't think you use `from_string` and I think you do run Jinja2 sandboxed, but confirmation would be useful

I don't think this is the right template but it's not a new vulnerability either :shrug: 

### Steps To Reproduce

any version of dbt core
install safety
run safety check

### Environment

```markdown
- OS:
- Python:
- dbt:
```

**Comment by dbeatty10** [comment]
Thanks for reaching out about this @brabster !

Sounds like you'd just like us to take a look and confirm that dbt-core is not affected by [CVE-2019-8341](https://github.com/advisories/GHSA-f6pv-j8mr-w6rr)?

Specifically:
- Confirm that dbt-core runs Jinja2 sandboxed
- If not, confirm that dbt-core does not use `from_string`

**Comment by brabster** [comment]
Yes please @dbeatty10 - I've done my own assessment (https://tempered.works/posts/2024/06/02/handling-cve-2019-8341-for-dbt-and-mkdocs/) but always helpful to get an all-clear and probably useful for others who might be getting pinged by safety as of Saturday!

**Comment by gshank** [comment]
Yes, we always subclass from SandboxedEnvironment.

**Comment by sfermigier** [comment]
Quoting https://bugzilla.redhat.com/show_bug.cgi?id=1677653#c4
"I'm one of the maintainers of the Pallets projects, including Jinja.
This CVE is a bad joke. It's like claiming eval() in the Python stdlib is insecure because it executes code.
Jinja templates should never be loaded from untrusted sources.
So nothing should be done here, there is literally nothing to be fixed."
